### PR TITLE
Vaurca Batch #2

### PR DESCRIPTION
**Pull request by LordFowl**
Adds a plethora of Vaurca changes along with a few other miscellaneous changes. Does NOT fix Vaurca surgery being broken.

Full details can be found in the changelog.

### CODE REVIEW

File: code/modules/mob/living/carbon/human/life.dm [hunk 1]
```diff
@@ -439,13 +439,42 @@
...
+			else
+				if(isnull(N) && all_languages["Hivenet"] in src.languages)
+					remove_language("Hivenet")
+					src << "\red Your mind suddenly grows dark as the unity of the Hive is torn from you."
+
+				if(N.is_broken() && all_languages["Hivenet"] in src.languages)
+					remove_language("Hivenet")
+					src << "\red Your mind suddenly grows dark as the unity of the Hive is torn from you."
+
+		if(l_hand && istype(l_hand, /obj/item/weapon/radio/vaurca) && !(src.get_species() == "Vaurca"))
```
Comment (skull132) [inline code/modules/mob/living/carbon/human/life.dm]: I'm dubious of this. Is there any better way to manage the assignment of this language, specially in cases of languages?

Perhaps look at how binary speech works with the binary headset/implant.

Also, the if/elseif/else structure here looks garbled. The else condition on line 473 only applies to the if on line 469. Ergo, it removes the language even if your left hand has a radio.

**Comment by LordFowl** [comment]
I'll take a look later tonight - I think you may be onto something.

File: code/modules/client/preferences_gear.dm [hunk 2]
```diff
@@ -1241,6 +1253,12 @@ var/global/list/gear_datums = list()
 	sort_category = "misc"
 	cost = 1
 
+/datum/gear/cape
```
Comment (skull132) [inline code/modules/client/preferences_gear.dm]: Erm, won't this give everyone a larger-than-average backpack? At the cost of almost nothing?

File: code/modules/mob/living/say.dm [hunk 3]
```diff
@@ -170,7 +170,11 @@ proc/get_radio_key_from_channel(var/channel)
 	if (speaking)
 		// This is broadcast to all mobs with the language,
 		// irrespective of distance or anything else.
-		if(speaking.flags & HIVEMIND)
+		if(speaking.flags & HIVEMIND && src.get_species() == "Vaurca")
+			speaking.broadcast(src,trim(message))
+			src << "\blue You consign your thoughts to the hive-net."
```
Comment (skull132) [inline code/modules/mob/living/say.dm]: In regular conversation, won't this get annoying as hell? To constantly have, "You consign your thoughts to the hive-net" sent every time you input a message?

File: code/modules/organs/organ_internal.dm [hunk 4]
```diff
@@ -167,17 +167,51 @@
...
+			owner.remove_language(all_languages["Hivenet"])
+			owner << "\red Your mind suddenly grows dark as the unity of the Hive is torn from you."
+	else
+		if (!(all_languages["Hivenet"] in owner.languages))
+			owner.add_language(all_languages["Hivenet"])
+			owner << "\blue Your mind expands, and your thoughts join the unity of the Hivenet."
+	..()
+
+/obj/item/organ/vaurca/neuralsocket/replaced(var/mob/living/carbon/human/target)
+	if(all_languages["Hivenet"] in target.languages)
```
Comment (skull132) [inline code/modules/organs/organ_internal.dm]: This is flipped. This needs to be:
```
	if(!(all_languages["Hivenet"] in target.languages))
 		target.add_language(all_languages["Hivenet"])
 		target << "\blue Your mind expands, and your thoughts join the unity of the Hivenet."
 	..()
```

File: code/modules/organs/organ_internal.dm [hunk 5]
```diff
@@ -167,17 +167,51 @@
...
+		target.remove_language(all_languages["Hivenet"])
+		target << "\red Your mind suddenly grows dark as the unity of the Hive is torn from you."
+	..()
+
+/obj/item/organ/vaurca/neuralsocket/removed(var/mob/living/carbon/human/target)
```
Comment (skull132) [inline code/modules/organs/organ_internal.dm]: And **this** needs to be:
```
    if(all_languages["Hivenet"] in target.languages)
        target.remove_language(all_languages["Hivenet"])
        target << "\red Your mind suddenly grows dark as the unity of the Hive is torn from you."
    ..()
```

File: code/modules/projectiles/gun.dm [hunk 6]
```diff
@@ -199,7 +199,25 @@
 		if(!(target && target.loc))
 			target = targloc
 			pointblank = 0
-
+	var/obj/projectile = consume_next_projectile(user)
```
Comment (skull132) [inline code/modules/projectiles/gun.dm]: Erm, what is this? And why were these modifications done?

File: code/modules/client/preferences_gear.dm (hunk 2 above)
Comment (ParadoxSpace) [inline code/modules/client/preferences_gear.dm]: I would hopefully assume it's whitelisted to vaurca usage only.

Comment (LordFowl) [inline code/modules/client/preferences_gear.dm]: It has a max_storage space of 12, which equates to about three items. Compare this to the backpack's max_storage space of 28.

File: code/modules/mob/living/say.dm (hunk 3 above)
Comment (LordFowl) [inline code/modules/mob/living/say.dm]: Perhaps. It's a relic from when hivenet was ability based. I've changed it now.

File: code/modules/organs/organ_internal.dm (hunk 4 above)
Comment (LordFowl) [inline code/modules/organs/organ_internal.dm]: Ok.

File: code/modules/projectiles/gun.dm (hunk 6 above)
Comment (LordFowl) [inline code/modules/projectiles/gun.dm]: These change how a gun displays its fire message and its feedback. Old-code wise a gun would display fire message and feedback whenever a projectile was created. This new revision makes it display fire message only once, when the user actually fires the gun. Audio and visual feedback is kept to the projectile. This reduces chatlog spam greatly when burst-fire weapons are employed.

**Pull request merged**